### DISCUSSION TITLE
Impersonation vs ContextSwitching

### DISCUSSION THREAD

**Discussion by cmaudoux** [nfrastack/container-lemonldap#29] [Q&A]
Hi Dave,

Congratulations for this docker file!
It is a very useful image. You have done an impressive work!

Just a comment, in a production system, prefer ContextSwitching plugin than Impersonation.
Impersonation has been developed to be used in training system to allow users testing another profiles...

With ContextSwitching you can allowed some users to switch context another users for debugging purpose.

Happy that you tested and deployed LemonLDAP::NG.

Best regards.

**Comment by tiredofit** [comment]
Thanks! I believe I was the person to make the feature request for Impersonation to be brought into the 2.x series back in the day (aka "spoofing" back then) - I think its important for users to know the difference in between them. In my production environment we actually use both for different use cases. 

I have added impersonation environment variables into this image due to some clever hacks for themes that we are using in production, but agreed, Most people should be using contextSwitching!

Lovely piece of software.
